text/0000-let-else.md

```diff
@@ -0,0 +1,520 @@
+because the compiler won't interpret it as `let PATTERN = (if y { a }) else { b };` since `()` is not an enum.
+
+This can be overcome by making a raw if-else in the expression position a compile error and instead requiring that parentheses are inserted to disambiguate:
+`let PATTERN = (if { a } else { b }) else { c };`.
+
+Rust already provides us with such a restriction, and so the expression can be restricted to be a [`ExpressionWithoutBlock`][expressions].
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+let-else attempts to be as consistent as possible to similar existing syntax.
+
+Fundamentally it is treated as a `let` statement, necessitating an assignment and the trailing semicolon.
+
+Pattern matching works identically to if-let, no new "negation" pattern matching rules are introduced.
+
+Operator precedence with `&&` in made to be like if-let, requiring that a case which is an error prior to this RFC be changed to be a slightly different error.
+This is for a possible extension for let-else similar to the (yet unimplemented) if-else-chains feature, as mentioned in [future-possibilities][] with more detail.
+Specifically, while the following example is an error today, by the default `&&` operator rules it would cause problems with if-let-chains like `&&` chaining:
+
+```rust
+let a = false;
+let b = false;
+
+// The RFC proposes boolean patterns with a lazy boolean operator (&& or ||)
+//  be made into a compile error, for potential future compatibility with if-let-chains.
+let true = a && b else {
+    return;
+};
+```
+
+The expression can be any [`ExpressionWithoutBlock`][expressions], in order to prevent `else {} else {}` confusion, as noted in [drawbacks][#drawbacks].
+
+The `else` must be followed by a block, as in `if {} else {}`. This else block must be diverging as the outer
+context cannot be guaranteed to continue soundly without assignment, and no alternate assignment syntax is provided.
+
+## Alternatives
+
+While this feature can partly be covered by functions such `or_or`/`ok_or_else` on the `Option` and `Result` types combined with the Try operator (`?`),
+such functions do not exist automatically on custom enum types and require non-obvious and non-trivial implementation, and may not be map-able
+to `Option`/`Result`-style functions at all (especially for enums where the "success" variant is contextual and there are many variants).
+These functions will also not work for code which wishes to return something other than `Option` or `Result`.
+
+### Naming of `else` (`let ... otherwise { ... }`)
+
+One often proposed alternative is to use a different keyword than `else`, such as `otherwise`.
+This is supposed to help disambiguate let-else statements from other code with blocks and `else`.
+
+This RFC avoids this as it would mean loosing symmetry with if-else & if-let-else, 
```

Comment:
```suggestion
This RFC avoids this as it would mean losing symmetry with if-else and if-let-else, 
```

Reply:
Thanks, I had seen this when reading the old RFC. Sorry I forgot about it.